contribs/gmf/src/controllers/AbstractDesktopController.js

```diff
@@ -246,11 +284,112 @@ export class AbstractDesktopController extends AbstractAPIController {
         })
       );
 
+    $scope.$watch(
+      () => this.dataPanelActive,
+      (newVal) => {
```

Comment:
It's somehow the same function as above in the file (104-115), can you make one define a single function to limit duplication of some lines ?